File: data/src/main/java/com/fishbrain/android/network/service/ApiService.java
package com.fishbrain.android.network.service;

import com.fishbrain.android.BuildConfig;
import com.fishbrain.android.domain.Catch;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;


/**
 * The RetrofitInstance service with the two available requests
 * <p/>
 * Created by dimitris.lachanas on 22/08/15.
 */
public interface ApiService {

	@GET(BuildConfig.CATCHES_URL)
	Call<List<Catch>> getCatches(@Query(BuildConfig.PAGE_URL) String pageId, @Query(BuildConfig.VERBOSITY_URL) String verbosity);

	@GET(BuildConfig.CATCHES_URL + "/{catchId}" + BuildConfig.VERBOSITY_URL)
	Call<Catch> getCatch(@Path("catchId") String catchId, @Query(BuildConfig.VERBOSITY_URL) int verbosity);

}

File: app/src/main/java/com/fishbrain/assignment/data/network/service/ApiService.kt
package com.fishbrain.assignment.data.network.service

import com.fishbrain.assignment.BuildConfig
import com.fishbrain.assignment.data.domain.Catch

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query


/**
 * The RetrofitInstance service with the two available requests
 *
 *
 * Created by dimitris.lachanas on 22/08/15.
 */
interface ApiService {

    @GET(BuildConfig.CATCHES_URL)
    fun getCatches(@Query(BuildConfig.PAGE_URL) pageId: String, @Query(BuildConfig.VERBOSITY_URL) verbosity: String): Call<List<Catch>>

    @GET(BuildConfig.CATCHES_URL + "/{catchId}" + BuildConfig.VERBOSITY_URL)
    fun getCatch(@Path("catchId") catchId: String, @Query(BuildConfig.VERBOSITY_URL) verbosity: Int): Call<Catch>

}